pcsx2/gui/Dialogs/GSDumpDialog.cpp

```diff
@@ -0,0 +1,806 @@
+	}
+}
+
+Dialogs::GSDumpDialog::GSThread::GSThread(GSDumpDialog* dlg)
+	: pxThread("GSDump")
+	, m_root_window(dlg)
+{
+}
+
+Dialogs::GSDumpDialog::GSThread::~GSThread()
+{
+	try
+	{
+		pxThread::Cancel();
+	}
+	DESTRUCTOR_CATCHALL
+}
+
+void Dialogs::GSDumpDialog::GSThread::OnStop()
+{
+	m_root_window->m_debug_mode->Disable();
+	m_root_window->m_start->Disable();
+	m_root_window->m_step->Disable();
+	m_root_window->m_selection->Disable();
+	m_root_window->m_vsync->Disable();
+	m_root_window->m_gif_list->DeleteAllItems();
+	m_root_window->m_gif_packet->DeleteAllItems();
+	m_root_window->m_gif_list->Refresh();
+	m_root_window->m_button_events.clear();
+}
+
+void Dialogs::GSDumpDialog::GSThread::ExecuteTaskInThread()
+{
+	pxInputStream dump_file(*m_root_window->m_selected_dump, new wxFFileInputStream(*m_root_window->m_selected_dump));
+
+	if (!dump_file.IsOk())
+	{
+		OnStop();
+		return;
+	}
+
+	u32 crc = 0, ss = 0;
+	// XXX: check the numbers are correct
+	int renderer_override = m_root_window->m_renderer_overrides->GetSelection();
+	char regs[8192];
+
+	dump_file.Read(&crc, 4);
+	dump_file.Read(&ss, 4);
+
+
+	char* state_data = (char*)malloc(sizeof(char) * ss);
+	dump_file.Read(state_data, ss);
+	dump_file.Read(&regs, 8192);
+
+	freezeData fd = {(int)ss, (s8*)state_data};
+	m_root_window->m_dump_packets.clear();
+
+	while (dump_file.Tell() < dump_file.Length())
+	{
+		GSType id = Transfer;
+		dump_file.Read(&id, 1);
+		switch (id)
+		{
+			case Transfer:
+			{
+				GSTransferPath id_transfer;
+				dump_file.Read(&id_transfer, 1);
+				s32 size = 0;
+				dump_file.Read(&size, 4);
+				char* transfer_data = (char*)malloc(size);
+				dump_file.Read(transfer_data, size);
+				GSData data = {id, transfer_data, size, id_transfer};
+				m_root_window->m_dump_packets.push_back(data);
+				break;
+			}
+			case VSync:
+			{
+				u8 vsync = 0;
+				dump_file.Read(&vsync, 1);
+				GSData data = {id, (char*)vsync, 1, Dummy};
+				m_root_window->m_dump_packets.push_back(data);
+				break;
+			}
+			case ReadFIFO2:
+			{
+				u32 fifo = 0;
+				dump_file.Read(&fifo, 4);
+				GSData data = {id, (char*)fifo, 4, Dummy};
+				m_root_window->m_dump_packets.push_back(data);
+				break;
+			}
+			case Registers:
+			{
+				char regs_tmp[8192];
+				dump_file.Read(&regs, 8192);
+				GSData data = {id, regs_tmp, 8192, Dummy};
+				m_root_window->m_dump_packets.push_back(data);
+				break;
+			}
+		}
+	}
+
+	if (m_root_window->m_debug_mode->GetValue())
+		m_root_window->GenPacketList(m_root_window->m_dump_packets);
+
+	return;
+	GetCorePlugins().Init();
+	GSsetBaseMem((void*)regs);
+	if (GSopen2((void*)pDsp, renderer_override) != 0)
+	{
+		OnStop();
+		return;
+	}
+
+	GSsetGameCRC((int)crc, 0);
+
+	if (!GetCorePlugins().DoFreeze(PluginId_GS, 0, &fd, true))
+		m_running = false;
+	GSvsync(1);
+	GSreset();
+	GSsetBaseMem((void*)regs);
+	GetCorePlugins().DoFreeze(PluginId_GS, 0, &fd, true);
+
+	size_t i = 0;
+	size_t RunTo = 0;
+	size_t debug_idx = 0;
```

Comment:
This and the 2 above can be moved further in.

Reply:
no they cannot, otherwise they'll be reinit to 0 at each iteration of the loop

Reply:
Ah right, forgot about that.